### PR TITLE
[BEAM-708] Using AutoValue in BoundedReadFromUnboundedSource

### PR DESCRIPTION
**Pull request by vectorijk**
Migrate BoundedReadFromUnboundedSource to use AutoValue to reduce boilerplate

### CODE REVIEW

**Comment by vectorijk** [comment]
R: @lukecwik @jbonofre

**Comment by jbonofre** [comment]
Thanks, I gonna take a look.

**Comment by davorbonaci** [comment]
Also, CC: @peihe.

File: sdks/java/core/src/main/java/org/apache/beam/sdk/io/BoundedReadFromUnboundedSource.java [hunk 1]
```diff
@@ -133,17 +140,27 @@ public void populateDisplayData(DisplayData.Builder builder) {
         .include("source", source);
   }
 
-  private static class UnboundedToBoundedSourceAdapter<T>
+  /**
+   * An Adapter wraps the underlying {@link UnboundedSource} with the specified bounds on
+   * number of records and read time into {@link BoundedSource}.
+   */
+  @AutoValue
+  public abstract static class UnboundedToBoundedSourceAdapter<T>
```
Comment (lukecwik) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/io/BoundedReadFromUnboundedSource.java]: package private?

File: sdks/java/core/src/main/java/org/apache/beam/sdk/io/BoundedReadFromUnboundedSource.java [hunk 2]
```diff
@@ -133,17 +140,27 @@ public void populateDisplayData(DisplayData.Builder builder) {
...
-    private UnboundedToBoundedSourceAdapter(
-        UnboundedSource<T, ?> source, long maxNumRecords, Duration maxReadTime) {
-      this.source = source;
-      this.maxNumRecords = maxNumRecords;
-      this.maxReadTime = maxReadTime;
+    @Nullable abstract UnboundedSource<T, ?> getSource();
+    @Nullable abstract long getMaxNumRecords();
+    @Nullable abstract Duration getMaxReadTime();
+
+    public abstract String toString();
```
Comment (lukecwik) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/io/BoundedReadFromUnboundedSource.java]: I don't think you need to declare toString()

File: sdks/java/core/src/main/java/org/apache/beam/sdk/io/BoundedReadFromUnboundedSource.java [hunk 3]
```diff
@@ -133,17 +140,27 @@ public void populateDisplayData(DisplayData.Builder builder) {
         .include("source", source);
   }
 
-  private static class UnboundedToBoundedSourceAdapter<T>
+  /**
+   * An Adapter wraps the underlying {@link UnboundedSource} with the specified bounds on
```
Comment (lukecwik) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/io/BoundedReadFromUnboundedSource.java]: Adapter wraps -> adapter that wraps

File: sdks/java/core/src/main/java/org/apache/beam/sdk/io/BoundedReadFromUnboundedSource.java [hunk 4]
```diff
@@ -133,17 +140,27 @@ public void populateDisplayData(DisplayData.Builder builder) {
...
+   * number of records and read time into {@link BoundedSource}.
```
Comment (lukecwik) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/io/BoundedReadFromUnboundedSource.java]: into {@link BoundedSource}. -> into a {@link BoundedSource}.

File: sdks/java/core/src/main/java/org/apache/beam/sdk/io/BoundedReadFromUnboundedSource.java (hunk 2 above)
Comment (vectorijk) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/io/BoundedReadFromUnboundedSource.java]: addressed.

File: sdks/java/core/src/main/java/org/apache/beam/sdk/io/BoundedReadFromUnboundedSource.java (hunk 1 above)
Comment (vectorijk) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/io/BoundedReadFromUnboundedSource.java]: Making it private will fail compilation. The reason would be Generated AutoValue annotation can't access private class here.

Comment (lukecwik) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/io/BoundedReadFromUnboundedSource.java]: I was suggesting package private which is represented by no visibility classifier:
"abstract static class"

**Comment by lukecwik** [comment]
LGTM
Will attempt to change visibility of UnboundedToBoundedSourceAdapter to package private before merging.

File: sdks/java/core/src/main/java/org/apache/beam/sdk/io/BoundedReadFromUnboundedSource.java (hunk 1 above)
Comment (vectorijk) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/io/BoundedReadFromUnboundedSource.java]: Sorry for the misunderstanding. (**Package** private). I will update it soon.

**Comment by lukecwik** [comment]
Thanks for your contribution. I added you as a contributor and assigned BEAM-708 to you and resolved it.

**Comment by vectorijk** [comment]
@lukecwik Thanks for that! Looking forward to contributing more.

**Pull request merged**